.github/workflows/cppcheck.yml

```diff
@@ -0,0 +1,31 @@
+---
+# SPDX-License-Identifier: BSD-3-Clause
+# Tools that can save round-trips to github and a lot of time:
+#
+# yamllint -f parsable this.yml
+# pip3 install ruamel.yaml.cmd
+# yaml merge-expand this.yml exp.yml && diff -w -u this.yml exp.yml
+#
+# github.com also has a powerful web editor that can be used without
+# committing.
+
+name: cppcheck
+
+# yamllint disable-line rule:truthy
+on: [pull_request, push]
+
+jobs:
+  cppcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        # From time to time this will catch a git tag and change SOF_VERSION
+        with: {fetch-depth: 50, submodules: recursive}
+
+      - name: apt install cppcheck
+        run: sudo apt-get -y install cppcheck
```

Comment:
You need `apt update` before any install unfortunately: https://github.com/actions/virtual-environments/issues/2924